projects/social/social.py

```diff
@@ -45,8 +47,30 @@ def populateGraph(self, numUsers, avgFriendships):
         self.users = {}
         self.friendships = {}
         # !!!! IMPLEMENT ME
+        if numUsers <= avgFriendships:
+            print('You need more users than avg friends')
+            return
 
+        totalFriendships = numUsers * avgFriendships
+        users_left = numUsers
+        friendship_counts = []
+        while totalFriendships != 0 and users_left != 0:
+            this_users_friendships = random.randint(0, totalFriendships)
+            users_left -= 1
+            friendship_counts.append(this_users_friendships)
+            print(this_users_friendships)
+            totalFriendships -= this_users_friendships
+        if len(friendship_counts) < numUsers:
+            remaining_users = numUsers - len(friendship_counts)
+            for i in range(0, remaining_users):
+                friendship_counts.append(0)
+        print(friendship_counts)
+        print(len(friendship_counts))
         # Add users
+        # their names are just numbers :shrugs:
+
+        for i in range(0, numUsers):
+            addUser(i)
 
         # Create friendships
 
```

Comment:
And then a hint for getting the social paths: 

I made a queue and I kept track of the path. While the queue was greater than 0 I would see if my path was in my visited set. 

Hopefully that helps. Let me know if you get stuck with the adventure game